packages/client-core/src/admin2/components/user/AddEditUserModal.tsx

```diff
@@ -0,0 +1,200 @@
+import Modal from '@etherealengine/ui/src/primitives/tailwind/Modal'
+import MultiSelect from '@etherealengine/ui/src/primitives/tailwind/MultiSelect'
+import Select, { SelectProps } from '@etherealengine/ui/src/primitives/tailwind/Select'
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import AccountIdentifiers from './AccountIdentifiers'
+
+const getDefaultErrors = () => ({
+  name: '',
+  avatarId: ''
+})
+
+export default function AddEditUserModal({ user }: { user?: UserType }) {
+  const { t } = useTranslation()
+
+  const userMutation = useMutation(userPath)
+  const avatarsQuery = useFind(avatarPath, {
+    query: {
+      action: 'admin'
+    }
+  })
+  const avatarOptions =
+    avatarsQuery.status === 'success'
+      ? [
+          { name: t('admin:components.user.selectAvatar'), value: '', disabled: true },
+          ...avatarsQuery.data.map((av) => ({ name: av.name, value: av.id }))
+        ]
+      : [{ name: t('common:select.fetching'), value: '', disabled: true }]
+
+  const scopeTypesQuery = useFind(scopeTypePath, {
+    query: {
+      paginate: false
+    }
+  })
+  const scopeTypeOptions: SelectProps['options'] =
+    scopeTypesQuery.status === 'success'
+      ? [
+          { name: t('admin:components.user.selectScopes'), value: '', disabled: true },
+          ...scopeTypesQuery.data.map((st) => ({ name: st.type, value: st.type }))
+        ]
+      : [{ name: t('common:select.fetching'), value: '', disabled: true }]
+
+  if (user) {
+    for (const scope of user.scopes || []) {
+      const scopeExists = scopeTypeOptions.find((st) => st.value === scope.type)
+      if (!scopeExists) {
+        scopeTypeOptions.push({ name: scope.type, value: scope.type })
+      }
+    }
+
+    if (!avatarOptions.find((av) => av.value === user.avatarId)) {
+      avatarOptions.push({ name: user.avatar.name || user.avatarId, value: user.avatarId })
+    }
+  }
+
+  const submitLoading = useHookstate(false)
+  const errors = useHookstate(getDefaultErrors())
+
+  const name = useHookstate(user?.name || '')
+  const avatarId = useHookstate(user?.avatarId || '')
+  const scopes = useHookstate(user?.scopes || [])
+
+  const handleSubmit = async () => {
+    if (!name.value) {
+      errors.name.set(t('admin:components.user.nameCantEmpty'))
+    }
+    if (!avatarId.value) {
+      errors.avatarId.set(t('admin:components.user.avatarCantEmpty'))
+    }
+    if (Object.values(errors.value).some((value) => value.length > 0)) {
+      return
+    }
+
+    const userData: UserData = {
+      name: name.value as UserName,
+      avatarId: avatarId.value as AvatarID,
+      isGuest: user?.isGuest,
+      scopes: scopes.value
+    }
+    submitLoading.set(true)
+    if (user?.id) {
+      await userMutation.patch(user.id, userData).then(() => PopoverState.hidePopupover())
+    } else {
+      await userMutation.create(userData).then(() => PopoverState.hidePopupover())
+    }
```

Comment:
- Why are we using both async/await and "then" at the same time?
- In case of errors, we should catch them, and show it in the modal without closing.

Reply:
corrected!